distribution/archives/build.gradle

```diff
@@ -44,6 +44,11 @@ task createPluginsDir(type: EmptyDirTask) {
   dir = "${pluginsDir}"
   dirMode = 0755
 }
+ext.jvmOptionsDir = new File(buildDir, 'jvm-options-hack/jvm.options.d')
```

Comment:
Should we include an example/readme/etc in here to make this a little more self-documenting and so folks don't think this is some kind of mistake? I guess the way I see it is eith a) we expect folks to read our documentation, at which point they'll just go ahead and create this folder or b) we don't, and we want to _lead_ them towards this capability, at which point we should probably provide a bit more context. Thoughts?

Reply:
We do sort them, see https://github.com/elastic/elasticsearch/pull/51882/files#diff-38516abb6f177c355f29699955e7e60eR72 in the implementation. However, this relies on the `Path#compareTo` implementation which explicitly says:

```
     * Compares two abstract paths lexicographically. The ordering defined by
     * this method is provider specific, and in the case of the default
     * provider, platform specific. This method does not access the file system
     * and neither file is required to exist.
```

"Provider" here refers to the filesystem provider. I am assuming (but did not investigate) this enables to account for odd sorting behavior such as on Windows via `StrCmpLogicalW` which sorts filenames by like `f-1`, `f-2`, `f-10` in the order `f-1`, `f-2`, `f-10` instead of `f-1`, `f-10`, `f-2`.

Reply:
Here's what I wanted to do: move the default heap options out of the root `jvm.options` into `jvm.options.d/heap.options`. We can't do this: it will break users that have provisioning in place to customize the heap size in the jvm.options file because `jvm.options.d/heap.options` would override this. It would break users badly, including Cloud.

One option here is to add a `jvm.options.d/heap.options` file with the heap size *commented out* and still remaining in `jvm.options` with a comment pointing to `jvm.optioins.d/heap.options`.

The primary reason that users customize JVM options is to customize the heap size. We have documentation on how to customize the heap size, which now points to `jvm.options.d`. My preference is that users consult this and the JVM options documentation as the single source for how to customize jvm.options.

>a) we expect folks to read our documentation, at which point they'll just go ahead and create this folder

This isn't necessary since the purpose of the hack that you're commenting on is to include this empty directory in the archive.

Thoughts?

Reply:
One downside of providing a `heap.options` file with the distribution is that it will mistakenly lead users to think that that file is special when in fact the heap options can be present in any file in `jvm.options.d`, and the last one wins. We might want to instead call this `custom.options`, if we go this route.

Reply:
> This isn't necessary since the purpose of the hack that you're commenting on is to include this empty directory in the archive.

Right, I get we want to create the empty directory, but without any content, it's purpose is a bit unclear. One option is just to put a `readme.txt` or similar in there to describe what the folder is for. Perhaps we could provide a simple snippet about setting heap options in there as an example. That way we can also make it clear that filenames are irrelevant.

Reply:
I'd rather not duplicate the documentation, our syntax is our own and users need to consult the documentation to understand it.